ir/instr.cpp

```diff
@@ -1865,6 +1985,28 @@ StateValue FpConversionOp::toSMT(State &s) const {
                : std::move(ret.value), np()};
   };
 
+  if (config::is_uf_float()) {
+    scalar = [&](const StateValue &sv, const Type &from_type,
+                 const Type &to_type) -> StateValue {
+      s.doesApproximation("uf_float", true);
+
+      ostringstream os;
+      os << getOpName() << "." << from_type << ".to." << to_type;
+      expr range = to_type.getDummyValue(true).value;
+      auto value = expr::mkUF(os.str(), {sv.value}, range);
+
+      AndExpr non_poison;
+      non_poison.add(sv.non_poison);
+      if (op != SIntToFP && !(op == UIntToFP && !(flags & NNEG)) &&
```

Comment:
what's this?

Reply:
Some of the floating point conversion operators are only partially defined (i.e. they return poison for some inputs), while others can convert all values and never return poison. The idea was to only generate the UF for the poison case if the conversion is partially defined. I've refactored this to use the `uf_float` function as well now. I think it should be much clearer now.